### PR TITLE
Add an example showing fading between two maps

### PR DESCRIPTION
**Pull request by dstansby**
Some code I had lying around, which might be a nice example for the gallery.

### CODE REVIEW

**Comment by wtbarnes** [comment]
I have no idea what's up with the docs build. I've just restarted it to see if that helps?

**Comment by wtbarnes** [comment]
I think you need to place the `add_axes` call and the `plot` calls all in the same block (i.e. not under different `#####` headings). Not entirely sure why this is, but this is what we're doing in all of our other examples.

**Comment by dstansby** [comment]
Thanks for the feedback - I've added some motivation text to the top of the example, but any suggestions on that are very welcome.

**Comment by wtbarnes** [comment]
Really a shame that the interactive output does not show up in the example gallery.

**Comment by Cadair** [comment]
> Really a shame that the interactive output does not show up in the example gallery.

That's because it's saving the plot before the slider and stuff is added.

**Comment by dstansby** [comment]
Thanks for the feedback.

> Really a shame that the interactive output does not show up in the example gallery.

Maybe this is a good excuse to build a bokeh plotting backend? 😛

**Comment by nabobalis** [comment]
> Thanks for the feedback.
> 
> > Really a shame that the interactive output does not show up in the example gallery.
> 
> Maybe this is a good excuse to build a bokeh plotting backend?

Don't you mean plotly?

**Comment by Cadair** [comment]
I think you need to refactor this due to https://github.com/sphinx-gallery/sphinx-gallery/issues/363

Specifically:

> [...] we are currently closing all the figures at the end of a code block. So unless I am wrong, the basic assumption is that a figure is completely built within a code block.

You can (and should) still add inline Python comments to explain the steps. This will mean that the figure actually displays as intended with the slider in the output.

**Comment by Cadair** [comment]
Also, good luck writing something equivalent to WCSAxes in either bokeh or plotly... :stuck_out_tongue_closed_eyes:

**Comment by dstansby** [comment]
I did combine them, so now the example looks like this: https://23569-2165383-gh.circle-artifacts.com/0/home/circleci/project/docs/_build/html/generated/gallery/plotting/fading_between_maps.html - I think this is fine, just needs some axes placement tweaking.

**Comment by Cadair** [comment]
I wish the API to do this with `sunpy.visualization.animator` was cleaner, it would do all the layout for you, but you have to subclass the thing.

File: examples/plotting/fading_between_maps.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE, AIA_1600_IMAGE
+
+###############################################################################
+# Start by loading two AIA maps from the sample data.
+#
+# Note that these maps must be co-aligned. If they are taken at different times
+# or from different observation locations they will need to be aligned using
+# rotate() or reproject().
```
Comment (Cadair) [inline examples/plotting/fading_between_maps.py]:
```suggestion
# `~sunpy.map.GenericMap.rotate` or `reproject`.
```

File: examples/plotting/fading_between_maps.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+"""
+Fading between two maps
```
Comment (nabobalis) [inline examples/plotting/fading_between_maps.py]:
```suggestion
=======================
Fading between two maps
```

File: examples/plotting/fading_between_maps.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE, AIA_1600_IMAGE
+
+###############################################################################
+# Start by loading two AIA maps from the sample data.
+#
+# Note that these maps must be co-aligned. If they are taken at different times
+# or from different observation locations they will need to be aligned using
+# `~sunpy.map.GenericMap.rotate` or `reproject`.
```
Comment (nabobalis) [inline examples/plotting/fading_between_maps.py]: Does reproject link now?

File: examples/plotting/fading_between_maps.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+map_171 = sunpy.map.Map(AIA_171_IMAGE)
```
Comment (nabobalis) [inline examples/plotting/fading_between_maps.py]:
```suggestion

map_171 = sunpy.map.Map(AIA_171_IMAGE)
```

File: examples/plotting/fading_between_maps.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+map_1600 = sunpy.map.Map(AIA_1600_IMAGE)
+
+###############################################################################
+# First we create a figure, and add the axes that will show the maps. Then
+# plot both of the images on the same axes. Finally, we add another axes that
+# contains the slider.
+fig = plt.figure()
```
Comment (nabobalis) [inline examples/plotting/fading_between_maps.py]:
```suggestion

fig = plt.figure()
```

File: examples/plotting/fading_between_maps.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+# Add the slider axes
+ax_slider = fig.add_axes([0.25, 0.05, 0.65, 0.03])
+slider = Slider(ax_slider, 'Alpha', 0, 1, valinit=0.5)
+
+
+###############################################################################
+# Finally, define what happens when the slider is changed and link this to the
+# slider we set up above. In this case we just set the alpha (ie. transparencey)
+# of the 171 image.
+def update(val):
```
Comment (nabobalis) [inline examples/plotting/fading_between_maps.py]:
```suggestion

def update(val):
```

File: examples/plotting/fading_between_maps.py (hunk 3 above)
Comment (Cadair) [reply examples/plotting/fading_between_maps.py]: It will post merge of the reproject example PR

**Pull request merged**